python/pyspark/sql/functions.py

```diff
@@ -11738,15 +11738,29 @@ def create_map(
 
     >>> from pyspark.sql import functions as sf
     >>> df = spark.createDataFrame([("Alice", 2, "female"),
-    ...                             ("Bob", 5, "male")], ("name", "age", "gender"))
+    ...     ("Bob", 5, "male")], ("name", "age", "gender"))
     >>> df.select(sf.create_map(sf.lit('name'), df['name'],
-    ...                         sf.lit('age'), df['age'])).show(truncate=False)
-    +-------------------------+
-    |map(name, name, age, age)|
-    +-------------------------+
-    |{name -> Alice, age -> 2}|
-    |{name -> Bob, age -> 5}  |
-    +-------------------------+
+    ...     sf.lit('gender'), df['gender'])).show(truncate=False)
```

Comment:
Did this example also fail in ANSI mode?

Reply:
I have manually tested both `export SPARK_ANSI_SQL_MODE=true` and `export SPARK_ANSI_SQL_MODE=false` cases locally
<img width="1420" alt="image" src="https://github.com/apache/spark/assets/15246973/12d89115-9104-45b0-bf1b-55ae72001a91">

Reply:
Will it fail before the modification?

Reply:
Yes, it will fail.
<img width="1419" alt="image" src="https://github.com/apache/spark/assets/15246973/9f090017-025c-418d-9512-455eea6c12ea">